### ISSUE TITLE
image_dataset_from_directory label list appear to be assigned in a random order

### ISSUE THREAD

**Issue by MichalKotecki** [keras-team/keras#14416]
I try to do the following:

I have 2 folders on Google Colab. First one contains PNG images and the second one contains CSV describing each image. The coresponding CSV files gave identical name to PNGs they describe, the only difference being the extension.
I go over all the PNG files. For each of them, I get the label from matching CSV and I build a label list.
The documentation (https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image_dataset_from_directory) states:
_Labels should be sorted according to the alphanumeric order of the image file paths (obtained via os.walk(directory) in Python)._

so I do exactly that:

```
label_list = []
for path, directories, image_files in os.walk(images_directory):
  for image_file in image_files:
    label_file = replace_file_extension(image_file, 'png', 'csv')
    label_list.append(get_class_label(SELECTED_FEATURE, label_file))

```
I use image_dataset_from_directory to create Dataset from directory with custom label list:
```
train_ds = tf.keras.preprocessing.image_dataset_from_directory(
    images_directory,
    image_size=(32,32),
    labels=label_list,
    label_mode="int"
    )
```
I get the labels assigned to my images in what appears a random order each time I load them. What is the reason behind this problem? How can I make sure images are loaded in a correct order?
